packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -434,27 +432,28 @@ abstract class FlameChartState<T extends FlameChart,
   }
 
   void _handleZoomControllerValueUpdate() {
-    setState(() {
-      final currentZoom = zoomController.value;
-      if (currentZoom == previousZoom) return;
+    final previousZoom = currentZoom;
+    currentZoom = zoomController.value;
+    if (previousZoom == currentZoom) return;
 
-      // Store current scroll values for re-calculating scroll location on zoom.
-      final lastScrollOffset = horizontalControllerGroup.offset;
 
-      final safeMouseHoverX = mouseHoverX ?? widget.containerWidth / 2;
-      // Position in the zoomable coordinate space that we want to keep fixed.
-      final fixedX = safeMouseHoverX + lastScrollOffset - widget.startInset;
+    // Store current scroll values for re-calculating scroll location on zoom.
+    final lastScrollOffset = horizontalControllerGroup.offset;
 
-      // Calculate the new horizontal scroll position.
-      final newScrollOffset = fixedX >= 0
-          ? fixedX * currentZoom / previousZoom +
-              widget.startInset -
-              safeMouseHoverX
-          // We are in the fixed portion of the window - no need to transform.
-          : lastScrollOffset;
+    final safeMouseHoverX = mouseHoverX ?? widget.containerWidth / 2;
+    // Position in the zoomable coordinate space that we want to keep fixed.
+    final fixedX = safeMouseHoverX + lastScrollOffset - widget.startInset;
 
-      previousZoom = currentZoom;
+    // Calculate the new horizontal scroll position.
+    final newScrollOffset = fixedX >= 0
+        ? fixedX * currentZoom / previousZoom +
+            widget.startInset -
+            safeMouseHoverX
+        // We are in the fixed portion of the window - no need to transform.
+        : lastScrollOffset;
 
+    setState(() {
+      currentZoom = zoomController.value;
```

Comment:
we already change currentZoom to this value outside of setState. I would avoid updating currentZoom until this line. Maybe add a 
nextZoom local variable to disambiguate.

Reply:
done